ddtrace/contrib/internal/flask/patch.py

```diff
@@ -110,25 +109,22 @@ class _FlaskWSGIMiddleware(_DDWSGIMiddlewareBase):
 
     def _wrapped_start_response(self, start_response, ctx, status_code, headers, exc_info=None):
         core.dispatch("flask.start_response.pre", (flask.request, ctx, config.flask, status_code, headers))
-        if not core.get_item(HTTP_REQUEST_BLOCKED):
-            headers_from_context = ""
-            result_waf = core.dispatch_with_results("flask.start_response", ("Flask",)).waf
-            if result_waf:
-                headers_from_context = result_waf.value
-            if core.get_item(HTTP_REQUEST_BLOCKED):
+        if not get_blocked():
+            core.dispatch("flask.start_response", ("Flask",))
+            if get_blocked():
```

Comment:
Same question here

Reply:
dispatching the appsec listener trigger a waf activation that can block the request. So basically here: if we did not already block the request, let's use the WAF (by core.dispatch). OK, and now, maybe we need to block the request.